beets/dbcore/db.py

```diff
@@ -26,8 +26,9 @@
 from collections import defaultdict
 from collections.abc import Generator, Iterable, Iterator, Mapping, Sequence
 from sqlite3 import Connection
-from typing import TYPE_CHECKING, Any, AnyStr, Callable, Generic, TypeVar
+from typing import TYPE_CHECKING, Any, AnyStr, Callable, Generic
 
+from typing_extensions import TypeVar  # default value support
```

Comment:
Hmmm, is this change needed? As far as I remember, it was about `mypy` supporting this from a certain __`mypy`__ version

Reply:
I think it is needed. We use `TypeVar(default=something)` some lines below which is only supported since version 3.13 in the std lib.

If the typevar line is needed is another question. I just assumed it is here for a good reason :)

Reply:
How come has it been working fine until now?

Reply:
I'm not convinced that this is required - let's continue importing it from `typing`. We can adjust it once we see any issues.

Reply:
https://docs.python.org/3/library/typing.html#typing.TypeVar see after has_default.

I would keep it as otherwise we need a if statement and this is way easier imo.

Reply:
It simplifies (i.e. removes the duplicated definition), other than that not really much value added. Sadly the `D` TypeVar cant be in the TYPE_CHECKING block as it is used as an value below (`class Model(ABC, Generic[D]):`).

Reply:
What about wrapping it with double quotes?

Reply:
Double quotes can go a long way with Python typehints, especially when it comes to forward references 😅 happy it works here, it means we can keep importing from `typing` - less lines to migrate later.